### PR TITLE
Use native fetch

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Instead of relying on `reqwest`, which has not been updated since 2015, we can use the native `fetch`, which is [widely supported](https://m.thegulocal.com/uk). We also already have a polyfill, for browser that do not support it natively, as IE11.

The documentation states the following differences between `lib/fetch` and window.fetch:

> - fetch
>   - input can only be a string, Request is not supported
>   - does not support JSONP
>   - headers can only be specified as an object literal, the Headers interface is not supported
> - Response
>   - blob is not supported
>   - formData is not supported
>   - headers are not populated

---

It seems that all these points are moot. Let’s go one by one:

>   - input can only be a string, Request is not supported
>   - headers can only be specified as an object literal, the Headers interface is not supported

Request and Headers are supported in fetch, [according to MDN](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API#fetch_interfaces)

---

>   - does not support JSONP

[JSONP](https://en.wikipedia.org/wiki/JSONP) is a thing of the past, isn’t it?

---

> - Response
>   - blob is not supported
>   - formData is not supported
>   - headers are not populated

`Response.blob()` is valid, and so is `.formData()`, [according to MDN](https://developer.mozilla.org/en-US/docs/Web/API/Body#methods).

Headers are coming through in the response, [again according to MDN](https://developer.mozilla.org/en-US/docs/Web/API/Response).

---

Follow-up from #23599.

## Does this change need to be reproduced in dotcom-rendering ?

- [X] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## What is the value of this and can you measure success?

Make code more modern and idiomatic. Reduce the size of the commercial bundle on DCR.

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [ ] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [ ] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [ ] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [X] Locally
- [x] On CODE (optional)

#### Step-by-step to test on `CODE`

- [ ] Comments load + expand signed out
- [ ] User can sign in
- [ ] Comments load + expand signed in
- [ ] Comments can be posted
- [ ] Comments can be replied to
- [ ] Comments can be recommended
- [ ] Check profile comment history
- [ ] comments can be picked
- [ ] comments can be reported
- [ ] individual comments can be linked (however, `CODE` discussion API permalinks to `PROD`)

#### Newsletter: `/email-newsletters`

- [X] Signed-out user can enter email and sign up
- [X] Signed-in user can sign up directly. 

#### Beacon `static/src/javascripts/projects/common/modules/analytics/beacon.js`

- [x] Beacon is removed

#### Lightbox 

- [ ] `static/src/javascripts/projects/common/modules/gallery/lightbox.js` ← could not test, as there is only ever one image.

#### Video metadata

- [ ] `static/src/javascripts/projects/common/modules/video/metadata.js` ← possibly present on old content, which uses `video-js.js`

#### Snaps (facia / font / thrashers?)

- [ ] `static/src/javascripts/projects/facia/modules/ui/snaps.js` ← could not find a snap with `data-snap-type` of `fragment`

#### fetchJson is used in the following files:

Manually checked files are shown below.

- [X] static/src/javascripts/bootstraps/enhanced/football.js
- [ ] static/src/javascripts/bootstraps/standard/main.js
- [x] static/src/javascripts/lib/fetch-json.ts:
- [ ] static/src/javascripts/lib/geolocation.js
- [ ] static/src/javascripts/projects/admin/bootstraps/radiator.js
- [ ] static/src/javascripts/projects/commercial/modules/hosted/next-video.js
- [ ] static/src/javascripts/projects/commercial/modules/hosted/onward.js
- [ ] static/src/javascripts/projects/common/modules/component.js
- [ ] static/src/javascripts/projects/common/modules/article/membership-events.js
- [ ] static/src/javascripts/projects/common/modules/article/rich-links.js
- [ ] static/src/javascripts/projects/common/modules/atoms/youtube.js
- [ ] static/src/javascripts/projects/common/modules/business/stocks.js
- [ ] static/src/javascripts/projects/common/modules/commercial/contributions-service.js
- [ ] static/src/javascripts/projects/common/modules/commercial/user-features.js
- [ ] static/src/javascripts/projects/common/modules/discussion/comment-count.js
- [ ] static/src/javascripts/projects/common/modules/discussion/comments.js
- [ ] static/src/javascripts/projects/common/modules/discussion/discussion-frontend.js
- [ ] static/src/javascripts/projects/common/modules/discussion/loader.js
- [ ] static/src/javascripts/projects/common/modules/discussion/whole-discussion.js
- [ ] static/src/javascripts/projects/common/modules/identity/api.ts:
- [ ] static/src/javascripts/projects/common/modules/identity/upsell/account-creation/AccountCreationForm.js
- [x] ~static/src/javascripts/projects/common/modules/onward/breaking-news.js~ removed in #23717
- [ ] static/src/javascripts/projects/common/modules/onward/related.js
- [ ] static/src/javascripts/projects/common/modules/social/share-count.js
- [X] static/src/javascripts/projects/common/modules/sport/football/match-info.js
- [ ] static/src/javascripts/projects/common/modules/sport/football/tag-page-stats.js
- [ ] static/src/javascripts/projects/common/modules/ui/autoupdate.js
- [ ] static/src/javascripts/projects/common/modules/video/metadata.js
- [ ] static/src/javascripts/projects/facia/modules/onwards/search-tool.js
- [ ] static/src/javascripts/projects/facia/modules/onwards/weather.js
- [ ] static/src/javascripts/projects/facia/modules/ui/container-show-more.js
- [ ] static/src/javascripts/projects/facia/modules/ui/live-blog-updates.js

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/analytics/beacon.js [hunk 1]
```diff
@@ -9,20 +8,4 @@ const fire = (path) => {
...
-
-    fetch(url, {
-        method: 'post',
-        header: {
-            'Content-Type': 'application/json',
-        },
-        body: jsonString,
-        mode: 'cors',
-    });
-};
```
Comment (mxdvl) [inline static/src/javascripts/projects/common/modules/analytics/beacon.js:26]: `postJson` is not used anywhere, and a rather trivial implementation.

cc. @guardian/ophan for visibility.

Comment (rtyley) [reply static/src/javascripts/projects/common/modules/analytics/beacon.js:26]: Hi, thanks for the cc- however, I don't think this code is related to Ophan - ie it's not used for writing data into Ophan's Tracker at ophan.theguardian.com - I think it's actually for monitoring ad-rendering, if I'm reading https://github.com/guardian/frontend/pull/17854/files correctly?

Comment (mxdvl) [reply static/src/javascripts/projects/common/modules/analytics/beacon.js:26]: Thanks, you’re right, and `performance-logging.js` was removed in #21652. Thanks @rtyley.

Comment (arelra) [reply static/src/javascripts/projects/common/modules/analytics/beacon.js:26]: Do we still need the `fire` function?

The only place its used is: [on-slot-render.js#L15](https://github.com/guardian/frontend/blob/main/static/src/javascripts/projects/commercial/modules/dfp/on-slot-render.js#L15)

Is it still useful?

Comment (mxdvl) [reply static/src/javascripts/projects/common/modules/analytics/beacon.js:26]: I’m proposing to remove it here: https://github.com/guardian/frontend/pull/23708

**Review by shtukas — APPROVED** [review]

**Pull request merged**